src/librustc/ty/layout.rs

```diff
@@ -213,6 +213,9 @@ impl<'a, 'tcx> LayoutCx<'tcx, TyCtxt<'a, 'tcx, 'tcx>> {
         let tcx = self.tcx;
         let param_env = self.param_env;
         let dl = self.data_layout();
+        let flat_pointer = Pointer(self.tcx.flat_addr_space());
+        let const_pointer = Pointer(self.tcx.const_addr_space());
```

Comment:
Why are these still present in layout? If we can always use the flat address space, why does a choice of address space have to leak into layouts?